apis/duck/verify.go

```diff
@@ -56,30 +56,53 @@ func VerifyType(instance interface{}, iface Implementable) error {
 	// that we will compare at the end.
 	input, output := iface.GetFullType(), iface.GetFullType()
 
+	if err := roundTrip(instance, input, output); err != nil {
+		return err
+	}
+
+	// Now verify that we were able to roundtrip all of our fields through the type
+	// we are checking.
+	if diff := cmp.Diff(input, output); diff != "" {
```

Comment:
Can you use Jon's safe version of this under `kmp`?

Reply:
Sure - was wondering if it didn’t use it for a reason